packages/react-server-dom-webpack/src/__tests__/ReactFlightDOM-test.js

```diff
@@ -623,6 +623,20 @@ describe('ReactFlightDOM', () => {
     );
   });
 
+  it('throws when await a client module prop of client exports', () => {
+    const ClientModule = clientExports({
+      Component: {deep: 'thing'},
+    });
+    async function awaitExport() {
+      const mod = await ClientModule
+      return await Promise.resolve(mod.Component)
+    }
+    expect(awaitExport).toThrowError(
```

Comment:
Does this throw synchronously or did you mean to 
```suggestion
    await expect(awaitExport()).rejects.toThrowError(
```
?